Add an images view to the column display

lsanwick opened the pull request:
This is revisiting a pull request a made a long time back, but in a much simpler manner.  Given there is already a column view, I just wanted to update it to display images instead of text labels.

I added an option to the setting to toggle which view to use.

![screen shot 2015-01-12 at 9 44 54 am](https://cloud.githubusercontent.com/assets/399536/5707722/f04a988a-9a3f-11e4-9874-0a9d9f5c2b41.png)

aeosynth commented:
the issue as before is small screens, where the text can be unreadable. what does this look like at 1280x1024?

lsanwick commented:
Looks fine actually, here's a screenshot. There is still the zoomed card image in the lower right as well, in case readability is an issue for someone.
![screen shot 2015-01-12 at 10 10 14 am](https://cloud.githubusercontent.com/assets/399536/5708139/70c6db60-9a43-11e4-8d8b-1b33f2dc0bf1.png)

lsanwick commented:
Any further thoughts here?

aeosynth commented on public/src/cards.js:
```diff
@@ -274,7 +274,7 @@ export function getZone(zoneName) {
   let groups = _.group(cards, sort)
   for (let key in groups)
-    _.sort(groups[key], 'name')
+    _.sort(groups[key], 'color', 'cmc', 'name')
```
sorting by multiple attributes should be a separate issue; it's unrelated to column images

aeosynth commented on public/src/components/settings.js:
```diff
@@ -60,6 +60,7 @@ export default React.createClass({
       this.Copy(),
       this.Side(),
       RBox('beep', 'beep for new packs'),
+      RBox('col_imgs', 'with images', {className: 'with_images'}),
```
javascript code typically uses camelCase, not under_scores

aeosynth commented on public/style.css:
```diff
@@ -29,6 +29,11 @@ time {
+  float: right;
+  margin-left: 5px;
+}
+
```
no

lsanwick replied:
The main reason I added this was to make it remain on the same line as the column view checkbox, so that it was clear that "with images" only applied to that view.  If I change it to be the default, this goes away.

aeosynth commented on public/src/components/game.js:
```diff
@@ -43,7 +43,7 @@ export default React.createClass({
       let pack = Grid({ zones: ['pack'] })
     let component = App.state.cols ? Cols : Grid
-    let pool = component({ zones: ['main', 'side', 'junk'] })
+    let pool = component({ zones: ['main', 'side', 'junk'], useImages: App.state.col_imgs })
```
i don't like passing props around, grab this setting inside the component

aeosynth commented:
honestly this could be the default, w/o a setting

lsanwick commented:
If you are happy with this being the default, I'll clean out the checkbox changes.  I was just being conservative given that the previous PR was rejected because it was changing the default behavior.

aeosynth commented:
your previous pr made unnecessary changes to the settings area, but it also had unresolved issues:

https://github.com/aeosynth/draft/pull/67#issuecomment-32548325

> the flicker is much more pronounced, and there is now a delay when mousing up the column; revert whatever you did here

> the card text is unreadable in my normal browser size; always show a full sized image

lsanwick commented:
Okay, removed the settings changes, and I've removed the sorting changes to do them in a separate request.

Made one minor bugfix to the new column view, which involved changing from onMouseEnter to onMouseOver.  If the commit message isn't clear, I can try and make a screencast to show the issue.

aeosynth commented on public/src/components/cols.js:
```diff
@@ -39,12 +39,12 @@ export default React.createClass({
-        d.div({
-          className: card.color,
+      let items = zone[key].map(card => {
+        return d.div({
```
you don't have to explicitly return from functions with a single expression:

https://github.com/lukehoban/es6features#arrows

aeosynth commented on public/src/components/cols.js:
```diff
@@ -39,12 +39,12 @@ export default React.createClass({
-          onMouseEnter: this.enter.bind(this, card.url)
-        }, card.name))
+          onMouseOver: this.enter.bind(this, card.url)
+        }, d.img({src: card.url}))
```
add spaces: `{ src: card.url }`

aeosynth commented on public/style.css:
```diff
@@ -61,6 +61,17 @@ time {
+}
+
+.zone .col div:last-child img {
+  margin-bottom: 0px;
```
this can be just `0`

aeosynth commented:
the code looks fine, but the card text is hard for me to read. increasing the size would reduce the number of columns visible at once, defeating the purpose of column mode. i'll have to think about this.

lsanwick commented:
Even in 1280x1024 the card text seemed fine to me, but my monitor physical size is about 15" What size are you viewing it on?

- Luke

> On Jan 14, 2015, at 5:54 PM, James Campos <notifications@github.com> wrote:
> 
> the code looks fine, but the card text is hard for me to read. increasing the size would reduce the number of columns visible at once, defeating the purpose of column mode. i'll have to think about this.
> 
> —
> Reply to this email directly or view it on GitHub.
>

aeosynth commented:
i have a 17" laptop, 1920x1200 resolution, browser width at 1280. i also have bad eyes. i might just release this and see if anyone complains, or maybe i will need an option for text mode. i went looking for screenshots of modo, and their text is also pretty bad. maybe an overlay of the name on top of the image, like cockatrice does.

lsanwick commented:
Yeah, MODO is awful, I wouldn’t use that as a basis for comparison.  I use a pretty ridiculously small font size, and even I have trouble reading cards on MODO.

- Luke Sanwick


On Wednesday, January 14, 2015 at 6:39 PM, James Campos wrote:

> i have a 17" laptop, 1920x1200 resolution, browser width at 1280. i also have bad eyes. i might just release this and see if anyone complains, or maybe i will need an option for text mode. i went looking for screenshots of modo, and their text is also pretty bad. maybe an overlay of the name on top of the image, like cockatrice does.
>  
> —
> Reply to this email directly or view it on GitHub (https://github.com/aeosynth/draft/pull/100#issuecomment-70031517).
>  
>  
>  
>  
>
